sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -138,6 +139,30 @@ class SQLAppStatusListener(
     }
   }
 
+  override def onStageCompleted(stageCompleted: SparkListenerStageCompleted): Unit = {
+    if (!isSQLStage(stageCompleted.stageInfo.stageId)) {
+      return
+    }
+    val stageNum = stageCompleted.stageInfo.stageId
+    val attemptID = stageCompleted.stageInfo.attemptNumber()
+
+    // gets the executionID that finished the stage
+    val liveExecution = liveExecutions.values().asScala
+    val execID = liveExecution.filter(_.stages.contains(stageNum)).head.executionId
```

Comment:
is this filter always returning non-empty collection ?

Reply:
Yes, I haven't seen a case that suggests otherwise.

Reply:
I'd prefer to see this check before assuming it is just in case something weird happens

Reply:
I added a check. Is it sufficient?